crates/blockifier/src/execution/contract_class.rs

```diff
@@ -122,20 +125,45 @@ pub struct ContractClassV0Inner {
     pub program: Program,
     pub entry_points_by_type: HashMap<EntryPointType, Vec<EntryPoint>>,
 }
+impl Encode for ContractClassV0Inner {
+    fn encode(&self) -> Vec<u8> {
+        let val = self.clone();
+        let entry_point_btree = val
+            .entry_points_by_type
+            .into_iter()
+            .collect::<BTreeMap<EntryPointType, Vec<EntryPoint>>>();
```

Comment:
You should replace `entry_points_by_type` type. Use indexmap instead of hashmap. Otherwise, you will end up with a different order of entries.